update docs to reflect main git enhancements (#4980)
fishtown-analytics/docs.getdbt.com: website/docs/docs/collaborate/git/version-control-basics.md

| Branch | A branch is a parallel version of a repository. It is contained within the repository but does not affect the primary or main branch allowing you to work freely without disrupting the live version. When you've made the changes you want to make, you can merge your branch back into the main branch to publish your changes |
| Checkout | The `checkout` command is used to create a new branch, change your current working branch to a different branch, or switch to a different version of a file from a different branch. |
| Commit | A commit is a user’s change to a file (or set of files). When you make a commit to save your work, Git creates a unique ID that allows you to keep a record of the specific changes committed along with who made them and when. Commits usually contain a commit message which is a brief description of what changes were made. |

> | main | The primary, base branch of all repositories. All committed and accepted changes should be on the main branch.<br /><br /> In the Cloud IDE, the main branch is protected. This means you can directly edit, format, or lint files and execute dbt commands in your protected primary git branch. When ready, you can commit those changes to a new branch. |

Author: ```suggestion
| main | The primary, base branch of all repositories. All committed and accepted changes should be on the main branch.<br /><br /> In the Cloud IDE, the main branch is protected. This means you can directly edit, format, or lint files and execute dbt commands in your protected primary git branch. Since the dbt Cloud IDE prevents commits to the protected branch, you can commit those changes to a new branch.|
```
